buyer-api/src/queues/notarizationQueue.js

```diff
@@ -1,11 +1,96 @@
+import uuid from 'uuid/v4';
 import { createQueue } from './createQueue';
-import { logger } from '../utils';
+import {
+  dataResponses,
+  dataResponsesBatches as batches,
+  notarizations,
+} from '../utils/stores';
+import logger from '../utils/logger';
+import config from '../../config';
+
+const { buyerPublicBaseUrl } = config;
+
+/**
+ * @async
+ * @function createNotarizationRequest
+ *  Creates a Notarization record containing the NotarizationRequest.
+ * @param {string} notaryAddress Notary's Ethereum address
+ * @param {number} orderId Order ID in the DataExchange contract
+ * @param {import('../utils/stores').NotarizationSeller[]} sellers List of sellers data
+ * @returns {string} ID of the Notarization record
+ */
+const createNotarizationRequest = (notaryAddress, orderId, sellers) => {
+  const id = uuid();
+  const callbackUrl = `${buyerPublicBaseUrl}/notarization-result/${id}`;
+  const request = {
+    orderId, sellers, callbackUrl, status: 'created',
+  };
+  notarizations.store(id, { notaryAddress, request });
+  return id;
+};
+
+/**
+ * @async
+ * @function collectNotarizationSellers
+ *  Fetches every DataResponse to extract the fields required for notarization.
+ * @param {string[]} dataResponseIds List of DataResponses IDs
+ * @returns {import('../utils/stores').NotarizationSeller[]} List of sellers data
+ */
+const collectNotarizationSellers = async dataResponseIds =>
+  dataResponseIds.reduce(async (accumulatorPromise, dataResponseId) => {
```

Comment:
This way it makes a loop of related `Promise`s.
We should make this in parallel since `DataResponse`s are not related to each other.
We can do this by using `await Promise.all` + `map`, and then `reduce` the `DataResponse[]`.

Reply:
This pattern is actually discouraged by eslint rules, but it can't lint it there since the repetition cycle is obfuscated by `reduce`.

Reply:
can you offer a alternative code? i fail to grasp your recommendation

Reply:
@Meketh We have many things to discuss here:
1) I don't find where it says that eslint discourages this pattern. In fact, I was going to suggest that we must start using this pattern for this cases. 
2) It is not obfuscated by reduce, it uses the behaviour of reduce to trasverse an array sequentially. 
3) eslint discourages for-like loops, not `reduce`.
4) I'm concerned by the amount of dataResponses we are going to handle. For development, there will be a few, but for production I think we can expect 1k at least. I'm not sure how a `Promise.all()` behaves with this amount of promises. Also, it would be interesting to see what happens with level at this point.
5) Since response time is not an issue at this point, I'd prefer to trasverse this sequentially rather than wait 1k or more `dataResponse.fetch`

Reply:
Actually I just realized this can be done in place in one line of code.
I'll add another comment on that line.

Reply:
I was talking about this rule: [no-await-in-loop](https://eslint.org/docs/rules/no-await-in-loop)